iib/workers/tasks/utils.py

```diff
@@ -1191,19 +1185,27 @@ def prepare_request_for_build(
 
     binary_image = build_request_config.binary_image(request_index, distribution_scope)
     if binary_image == "scratch":  # binaryless image mode
+        arches = set(["amd64"])
         binary_image_resolved = binary_image
         binary_image_arches = arches
     else:
+        arches = gather_index_image_arches(build_request_config, index_info)
         binary_image_resolved = get_resolved_image(binary_image)
         binary_image_arches = get_image_arches(binary_image_resolved)
 
+    if not arches:
```

Comment:
This already duplicates the check in `get_image_arches` and when I look at the code I think it might not be necessary to have it here.

Reply:
Sorry, I don't get it..

The `get_image_arches` has only a check if the manifest is `V2` or not, if there are no arches it will return an empty set... Or am I missing something?

Reply:
Yeah, my bad - I meant `gather_index_image_arches` is raising the same. (as Michal mentioned belov)

Reply:
Cool! I'll get rid of it then, tks!